include/envoy/http/filter.h

```diff
@@ -512,6 +521,15 @@ class StreamEncoderFilter : public StreamFilterBase {
    */
   virtual FilterTrailersStatus encodeTrailers(HeaderMap& trailers) PURE;
 
+  /**
+   * Called with metadata to be encoded. New metadata should be added directly to metadata_map. DO
+   * NOT call StreamDecoderFilterCallbacks::encodeMetadata() interface to add new metadata.
```

Comment:
I'm confused, isn't it OK to call this from a decoder filter, just not an encoder filter? Should that function be removed from callbacks perhaps if we don't want people to call it?

Reply:
Got it, OK makes sense.